Automattic/themes
# Varia + Hever: Refactor color annotations

#### Changes proposed in this Pull Request:

This PR refactors the color annotations so that most of the code lives on Varia while maintaining the ability to leverage either of the two methods by its children.

This PR:

* Moves the IE stylesheet to a separate stylesheet that is not compiled by the children but loaded only when it's needed.
* Defines the editor color palette within Varia. The children using the old method can still redefine the palette the old way and the children using the new method can avoid doing so by defining the colors initially within `varia_default_colors`
* Creates a secondary file `wpcom-editor-colors.php` with the new implementation of the color annotation that should be imported only by the children that are using it and ignored by the rest (including Varia).
* Moves all the JS color utilities to Varia instead of having them on the children that implement de CSS variable method.

To test:

* Check that Varia color annotations work ~~fine~~ **the same way they did before this patch** (and one of the children with the old implementation for good measure)
* Check that the editor color palette is still working ~~as expected~~ **the same way it did before this patch** and changes when the colors are changed.
* Do the same with Hever.

This PR will require a full build after it's approved, but I'm holding on for clarity of the code.
